app/routes/password/reset.js

```diff
@@ -24,10 +24,10 @@ export default Route.extend({
         data: {
           token: get(this, 'controller.token'),
           password,
-          passwordConfirmation
+          "password-confirmation": passwordConfirmation
```

Comment:
It's because most of the keys in JSON API are dasherized by default. There's some lack of consistency in some places at present that are not standard CRUD (like here), but would like to bring at least that consistency to bear here, even if the resource itself isn't a JSON API resource.